scripts/build.mjs

```diff
@@ -0,0 +1,228 @@
+          }
+        });
+      },
+    },
+  ],
+  resolve: {
+    alias: [
+      { find: '/hybrids.js', replacement: 'hybrids' },
+      { find: '/vendor', replacement: '/src/vendor' },
+    ],
+  },
+  build: {
+    outDir: options.outDir,
+    emptyOutDir: false,
+    minify: false,
+    polyfillModulePreload: false,
+    watch: argv.watch ? {} : null,
+  },
+};
+
+// --- Generate dist structure ---
+
+// generate dist folder
+execSync(`rm -rf ${options.outDir} && mkdir -p ${options.outDir}`);
+
+// copy static assets
+options.assets.forEach((path) => {
+  execSync(`mkdir -p ${options.outDir}/${path}`);
+  execSync(`cp -r ${options.srcDir}/${path}/* ${options.outDir}/${path}`);
+});
+
+// Save manifest
+writeFileSync(
+  resolve(options.outDir, 'manifest.json'),
+  JSON.stringify(manifest, null, 2),
+);
+
+// --- Generate entry points ---
+
+const source = [];
+const content_scripts = [];
+
+if (manifest.action?.default_popup) {
+  source.push(manifest.action.default_popup);
+}
+
+if (manifest.browser_action?.default_popup) {
+  source.push(manifest.browser_action.default_popup);
+}
+
+// options page
+if (manifest.options_page) {
+  source.push(manifest.options_page);
+}
+
+// content scripts
+manifest.content_scripts?.forEach(({ js = [], css = [] }) => {
+  [...js, ...css].forEach((src) => content_scripts.push(src));
+});
+
+// web-accessible resources
+manifest.web_accessible_resources?.forEach((entry) => {
+  if (typeof entry === 'string') {
+    source.push(entry);
+  } else {
+    entry.resources.forEach((src) => source.push(src));
+  }
+});
+
+// background
+if (manifest.background) {
+  source.push(manifest.background.service_worker || manifest.background.page);
+}
+
+// --- Build  ---
+
+function mapPaths(paths) {
+  return paths.reduce((acc, src) => {
+    acc[src] = resolve(
+      src.startsWith('node_modules') ? src : resolve(options.srcDir, src),
+    );
+    return acc;
+  }, {});
+}
+
+await build({
+  ...config,
+  build: {
+    ...config.build,
+    target: 'esnext',
+    rollupOptions: {
+      input: mapPaths(source),
+      preserveEntrySignatures: 'exports-only',
+      output: {
+        dir: options.outDir,
+        manualChunks: false,
+        preserveModules: true,
+        preserveModulesRoot: 'src',
+        minifyInternalExports: false,
+        entryFileNames: (chunkInfo) => {
+          if (chunkInfo.facadeModuleId.endsWith('.css')) {
+            return 'common/[name].css';
+          }
+          if (chunkInfo.isEntry) {
+            return chunkInfo.facadeModuleId.replace(/^.*\//, '');
+          }
+          return '[name].js';
+        },
+        assetFileNames: (chunkInfo) => {
+          return chunkInfo.name.includes('/') && chunkInfo.name.endsWith('.css')
+            ? chunkInfo.name.replace('.css', '')
+            : 'assets/[name].[ext]';
+        },
+      },
+    },
+  },
+});
+
+// --- Build content scripts ---
+
+for (const [key, value] of Object.entries(mapPaths(content_scripts))) {
```

Comment:
maybe call `key` and `value` differently, for example `entryPointName` and `entryPointPath`